### PR TITLE
Removed comma separator

### LINKED ISSUE #5260: [Bug]: Descriptions in multiple languages should not be separated by comma
### Summary

The `description` parameter in the `{{Information}}` template filled by the application when uploading an image contains description in provided languages, _separated by commas_ (`,`). Which is wrong. The individual `{{en|...}}`, `{{de|...}}` templates used in the description are rendered as blocks, meaning the separating commas are rendered between lines of individual languages, looking terrible.

See e.g. [this upload](https://commons.wikimedia.org/w/index.php?title=File:Cape_Donji_lighthouse_in_Slano.jpg&oldid=781651936).

The languages can be separated by whitespace (e.g. newlines) or not at all.

I believe this error is at `fr.free.nrw.commons.description.DescriptionEditActivity#updateDescription`. Note that the same logic is at multiple places in the application, including `fr.free.nrw.commons.upload#formatList` and `fr.free.nrw.commons.contributions#formatDescriptions`, implemented in a different style every time, but those two are correct and do not use any separator.

### Steps to reproduce

1. Upload an image with descriptions in more than one language.
2. Take a look at the resulting description page.

### Expected behaviour

The descriptions should look fine.

### Actual behaviour

There is a stray comma between every two language versions in the description.

### Device name

Samsung S22

### Android version

Android 13

### Commons app version

4.1.0~1649d1e2c

### Device logs

_No response_

### Screen-shots

_No response_

### Would you like to work on the issue?

None

### PR DESCRIPTION
**Pull request by paco-arana**
**Description**

Fixes #5260

area: `fr.free.nrw.commons.description.DescriptionEditActivity#updateDescription`

The comma separator between descriptions in different languages has been removed from `fr.free.nrw.commons.description.DescriptionEditActivity#updateDescription` line 154. This also made line 157 redundant as its function was deleting the very last comma after all the descriptions were formatted.

**Tests performed**

Tested beta build on Xiaomi Redmi 9 using Android 11.

### CODE REVIEW

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I uploaded a file using your branch and still see a comma, that's strange: https://commons.wikimedia.org/wiki/File:Binh_Tay_market_2.jpg

Maybe DescriptionEditActivity is only used when modifying an existing file, rather than when uploading a new file?

Thanks! :-)

File: app/src/main/java/fr/free/nrw/commons/contributions/Contribution.kt [hunk 1]
```diff
@@ -112,5 +112,6 @@ data class Contribution constructor(
         fun formatDescriptions(descriptions: List<UploadMediaDetail>) =
             descriptions.filter { it.descriptionText.isNotEmpty() }
                 .joinToString { "{{${it.languageCode}|1=${it.descriptionText}}}" }
+                .replace("}}, ", "}}")
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/Contribution.kt:115]: Would not it be better to not add the comma in the first place?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/contributions/Contribution.kt (hunk 1 above)
Comment (paco-arana) [reply app/src/main/java/fr/free/nrw/commons/contributions/Contribution.kt:115]: I tried that first but I was having a hard time determining where the comma was actually originating from, I'll dig a little more until I can find it. 

Thanks for the quick response, I apologize if I'm making you work more than needed.

Comment (paco-arana) [reply app/src/main/java/fr/free/nrw/commons/contributions/Contribution.kt]: Okay, I think this one should do it, I should've checked the default separator for the .joinToString first before trying to look for the comma everywhere else.

**Review by nicolas-raoul — COMMENTED** [review]
Looks great! I will test your pull request as soon as possible. 🙂

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Uploading now works great! :-)
Unfortunately editing adds the commas:
https://commons.wikimedia.org/w/index.php?title=File%3AKaba_amphibious_bus_in_Odaiba.jpg&diff=784750015&oldid=784749122
Would you mind checking?

**Comment by paco-arana** [comment]
I seem to have caused that myself when I modified DescriptionEditActivity#updateDescription, I added the deleted line back in and now it's working as it should:

https://commons.wikimedia.org/w/index.php?title=File:Catedral_de_Aguascalientes.jpg&diff=prev&oldid=784872825

**Review by nicolas-raoul — APPROVED** [review]
Now working great for both upload and modification.
Example: https://commons.wikimedia.org/wiki/File:Central_Building,_Sakura_Tower.jpg
Thanks a lot!

**Pull request merged**